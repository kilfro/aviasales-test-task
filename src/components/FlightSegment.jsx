import { getDurationString, getFlightTimes, getStopsQuantityString } from '../functions'

import React from 'react'

const FlightSegment = ({ origin, destination, date, duration, stops }) => (
    <div className='flight-segment'>
        <div className='segment-part'>
            <div>{origin} - {destination}</div>
            <div>{getFlightTimes(new Date(date), duration)}</div>
        </div>
        <div className='segment-part'>
            <div>в пути</div>
            <div>{getDurationString(duration)}</div>
        </div>
        <div className='segment-part'>
            <div>{getStopsQuantityString(stops.length)}</div>
            <div>{stops.join(', ')}</div>
        </div>
    </div>
)

export default FlightSegment