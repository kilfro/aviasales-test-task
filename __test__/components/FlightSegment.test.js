import FlightSegment from '../../src/components/FlightSegment'
import React from 'react'

describe(FlightSegment, () => {
  test('must be rendered correctly', () => {
    const props = {
      origin: 'SVO',
      destination: 'LED',
      date: new Date(2021, 3, 7, 19, 45).toString(),
      duration: 75,
      stops: ['SVX'],
    }
    const component = shallow(<FlightSegment {...props} />)
    expect(component).toMatchSnapshot()
  })
})
