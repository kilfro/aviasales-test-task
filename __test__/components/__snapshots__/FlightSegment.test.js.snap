// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightSegment must be rendered correctly 1`] = `
<div
  className="flight-segment"
>
  <div
    className="segment-part"
  >
    <div>
      SVO
       - 
      LED
    </div>
    <div>
      19:45 - 21:00
    </div>
  </div>
  <div
    className="segment-part"
  >
    <div>
      в пути
    </div>
    <div>
      1ч 15м
    </div>
  </div>
  <div
    className="segment-part"
  >
    <div>
      1 пересадка
    </div>
    <div>
      SVX
    </div>
  </div>
</div>
`;
